File: app/src/main/java/io/scribeapp/classifier/model/ClassificationRequestSerializer.java
package io.scribeapp.classifier.model;

import android.util.Log;

import com.google.gson.JsonElement;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import java.lang.reflect.Type;

public class ClassificationRequestSerializer implements JsonSerializer<ClassificationRequest> {

    @Override
    public JsonElement serialize(ClassificationRequest src, Type typeOfSrc, JsonSerializationContext context) {
        JsonElement jsonElement = src.toJson();
        Log.d("JSON", jsonElement.toString());
        return jsonElement;
    }
}

File: app/src/main/kotlin/io/scribe/classifier/model/ClassificationRequestSerializer.kt
package io.scribe.classifier.model

import com.google.gson.JsonSerializer
import com.google.gson.JsonElement
import com.google.gson.JsonSerializationContext
import java.lang.reflect.Type

public class ClassificationRequestSerializer : JsonSerializer<ClassificationRequest> {

    override fun serialize(src: ClassificationRequest, typeOfSrc: Type, context: JsonSerializationContext): JsonElement {
        return src.toJson()
    }
}